File: src/logicServer/src/main/java/via/sep3/logicserver/shared/Workout.java
package via.sep3.logicserver.shared;

import java.util.List;

import org.springframework.stereotype.Component;

import lombok.Data;

@Data
@Component
public class Workout {
    private int id;
    private String Name;
    private String Description;
    private int DurationInMin;
    private int CreatedBy;
    private int FollowedBy;
    private boolean isPublic;
    private List<Exercise> Exercises;

    public int numberOfExercises() {
        return Exercises.size();
    }
}

File: src/dataAccessServer/Shared/DTOs/Workout.cs
namespace Shared.DTOs;

public class Workout
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int DurationInMin { get; set; }
    public int NumberOfExercises()
    {
        return Exercises.Count;
    }
    public int CreatedBy { get; set; }
    public int FollowedBy { get; set; }
    public bool IsPublic { get; set; }
    public List<Exercise> Exercises { get; set; }
}